### PR TITLE
Warrior baseline spells

### PR DESCRIPTION
**Pull request by Swillis57**
Well... I tried to keep it small.

Adds warrior spells shared across specs. Simple talents that affected said spells were also implemented, but anything more involved I'll handle as I'm implementing the talent trees and glyphs.

### CODE REVIEW

**Comment by Swillis57** [comment]
This includes changes already in https://github.com/wowsims/cata/pull/37 so that should be merged first.

**Review by kayla-glick — APPROVED** [review]

File: sim/warrior/thunder_clap.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	warrior.ThunderClap = warrior.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 6343},
+		SpellSchool:    core.SpellSchoolPhysical,
+		ProcMask:       core.ProcMaskRangedSpecial,
+		Flags:          core.SpellFlagIncludeTargetBonusDamage | core.SpellFlagAPL,
+		ClassSpellMask: SpellMaskThunderClap,
+
+		RageCost: core.RageCostOptions{
+			Cost: 20 - core.TernaryFloat64(warrior.HasMajorGlyph(proto.WarriorMajorGlyph_GlyphOfResonatingPower), 5, 0),
```
Comment (rosenrusinov) [inline sim/warrior/thunder_clap.go:23]: Can this be moved to the static spell mods?

File: sim/warrior/overpower.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+	"github.com/wowsims/cata/sim/core"
+)
+
+func (warrior *Warrior) RegisterOverpowerSpell() {
+	opAura := warrior.RegisterAura(core.Aura{
+		ActionID: core.ActionID{SpellID: 7384},
+		Label:    "Overpower Ready",
+		Duration: time.Second * 5,
+	})
```
Comment (rosenrusinov) [inline sim/warrior/overpower.go:14]: need to tag it with EnableOverpowerTag?

**Review by rosenrusinov — APPROVED** [review]

File: sim/warrior/thunder_clap.go (hunk 1 above)
Comment (Swillis57) [reply sim/warrior/thunder_clap.go:23]: Oh it's already in there, forgot to remove this

**Pull request merged**